src/NUnitEngine/nunit.engine.core/ITestAgent.cs

```diff
@@ -21,26 +21,13 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 // ***********************************************************************
 
-using System;
-
 namespace NUnit.Engine
 {
     /// <summary>
     /// The ITestAgent interface is implemented by remote test agents.
     /// </summary>
     public interface ITestAgent
     {
-        /// <summary>
-        /// Gets a Guid that uniquely identifies this agent.
-        /// </summary>
-        Guid Id { get; }
-
-        /// <summary>
-        /// Starts the agent, performing any required initialization
-        /// </summary>
-        /// <returns>True if successful, otherwise false</returns>
-        bool Start();
```

Comment:
Do I understand correctly that all public engine types are changeable in every release because consumers of the engine should only be referencing the engine API assembly?

Reply:
Yes, that's my opinion. I want to enforce that in v4. 🙂

Reply:
In V3, we could at least tell people this.

Reply:
A fair point. I went to take a look at what our current docs say. I think the first paragraph of this page covers that the API is the supported way to integrate with the engine? 

https://github.com/nunit/docs/wiki/Test-Engine-API

I'd love to spin out the engine and console docs at some point, and build a little more structure around them. One day...

Reply:
A fair point. I went to take a look at what our current docs say. I think the first paragraph of this page covers that the API is the supported way to integrate with the engine? 

https://github.com/nunit/docs/wiki/Test-Engine-API

I'd love to spin out the engine and console docs at some point, and build a little more structure around them. One day...

Reply:
@ChrisMaddock your last comment showed up twice, so I deleted one of the copies.

Reply:
I think that doc page says it well. I was only reacting to the "modern" view sometimes expressed that any public method being changed constitutes a breaking change.